src/codegen.cpp

```diff
@@ -1796,8 +1794,8 @@ static void jl_setup_module(Module *m, const jl_cgparams_t *params = &jl_default
     if (!m->getModuleFlag("Debug Info Version"))
         m->addModuleFlag(llvm::Module::Warning, "Debug Info Version",
             llvm::DEBUG_METADATA_VERSION);
-    m->setDataLayout(jl_data_layout);
-    m->setTargetTriple(jl_TargetMachine->getTargetTriple().str());
+    m->setDataLayout(jl_ExecutionEngine->getDataLayout());
+    m->setTargetTriple(jl_ExecutionEngine->getTargetTriple().str());
```

Comment:
If we could pass this as an argument, it would also save us from needing to mutate it later in `jl_dump_native_impl`

This function could also be fused into `_jl_create_llvm_module`

Reply:
Can we also switch this to taking the data layout and triple as arguments (perhaps as part of a `jl_codegen_params_t` argument)?

Reply:
It looks like the code in jl_dump_native_impl just adds the GC pointer address spaces to the data layout string, which should now be stored in jl_ExecutionEngine->getDataLayout() instead. Assuming that we're getting the module from jl_create_llvm_module, the data layout of the module shouldn't need to be reset from then in jl_dump_native_impl. Would it be better to simply replace these lines with asserts rather than set calls?

Reply:
Eventually we will probably want to avoid asserting anything about the layout, and let the caller set it. But eventually we probably will just stop having codegen make modules, and just expect the caller to provide it fully initialized.